scripts/esbuild/util.ts

```diff
@@ -0,0 +1,88 @@
+import { join } from 'path';
+
+import { BuildOptions } from '../utils/options';
+
+/**
+ * Aliases which map the module identifiers we set in `paths` in `tsconfig.json` to
+ * bundles (built either with esbuild or with rollup).
+ *
+ * @param opts options for the current build
+ * @returns a map of aliases to bundle entry points, relative to the root directory
+ */
+export function getEsbuildAliases(opts: BuildOptions): Record<string, string> {
+  return {
+    // node module redirection
+    chalk: 'ansi-colors',
+
+    // mapping aliases to top-level bundle entrypoints
+    '@stencil/core/testing': './testing/index.js',
+    '@stencil/core/compiler': './compiler/stencil.js',
+    '@stencil/core/dev-server': './dev-server/index.js',
+    '@stencil/core/mock-doc': './mock-doc/index.cjs',
+    '@stencil/core/internal/testing': './internal/testing/index.js',
+
+    // this has to be added here pointing to the `.ts` file so that the `import
+    // { IS_NODE_ENV } from '@environment';` in the `typescript.js` file we
+    // bundle will resolve correctly
+    //
+    // esbuild will automatically resolve aliases from `paths` in
+    // `tsconfig.json` but only for `.ts` files, so when we bundle `typescript`
+    // into e.g. the compiler bundle we'll need this so things resolve
+    // correctly
+    '@environment': join(opts.srcDir, 'compiler/sys/environment.ts'),
```

Comment:
Do we still need this, now that we removed in-browser compilation? I see some usages of `@environment`, but I _think_ those are all infrastructure (and not actual imports that use the alias).

Reply:
oh yeah we probably can get rid of that! I think there might be one or two other things that can be simplified. I'll do a bit of an audit for that